### PR TITLE
Fix style and type issues

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Mypy checks just got stricter it seems, and rightfully so!

Edit: fixed some style issues as well

List of type-checking issues from mypy 1.1.1

```
src/gluonts/evaluation/metrics.py:90: error: Incompatible return value type (got "floating[Any]", expected "float")  [return-value]
src/gluonts/zebras/_period.py:52: error: Library stubs not installed for "dateutil.parser"  [import]
src/gluonts/zebras/_period.py:52: note: Hint: "python3 -m pip install types-python-dateutil"
src/gluonts/zebras/_period.py:52: note: (or run "mypy --install-types" to install all missing stub packages)
src/gluonts/zebras/_period.py:52: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/gluonts/dataset/repository/_airpassengers.py:44: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/dataset/artificial/_base.py:168: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/dataset/artificial/_base.py:171: error: List item 0 has incompatible type "Dict[str, str]"; expected "CategoricalFeatureInfo"  [list-item]
src/gluonts/dataset/artificial/_base.py:176: error: List item 0 has incompatible type "Dict[str, str]"; expected "BasicFeatureInfo"  [list-item]
src/gluonts/dataset/artificial/_base.py:180: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/dataset/artificial/_base.py:183: error: List item 0 has incompatible type "Dict[str, str]"; expected "CategoricalFeatureInfo"  [list-item]
src/gluonts/dataset/artificial/_base.py:188: error: List item 0 has incompatible type "Dict[str, str]"; expected "BasicFeatureInfo"  [list-item]
src/gluonts/dataset/artificial/_base.py:447: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/dataset/artificial/_base.py:805: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/dataset/artificial/_base.py:812: error: Argument "cardinality" to "CategoricalFeatureInfo" has incompatible type "int"; expected "str"  [arg-type]
src/gluonts/dataset/artificial/_base.py:836: error: Unexpected keyword argument "freq" for "MetaData"  [call-arg]
src/gluonts/torch/model/simple_feedforward/module.py:99: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:161: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:165: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:173: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:177: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:181: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/torch/model/deepar/module.py:189: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
src/gluonts/mx/trainer/learning_rate_scheduler.py:250: error: Too many positional arguments for "Patience"  [misc]
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
Made this breaking because of the change to `MetaData`

File: src/gluonts/zebras/_period.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 from typing import Any, Union, cast, overload
 
 import numpy as np
-from dateutil.parser import parse as du_parse
+from dateutil.parser import parse as du_parse  # type: ignore
```
Comment (jaheba) [inline src/gluonts/zebras/_period.py]: What is this one about?

File: src/gluonts/model/inputs.py [hunk 2]
```diff
@@ -14,15 +14,15 @@
...
+from typing import Any, Callable, Dict, Type, Tuple
 
 from toolz import valmap
 
 
 @dataclass
 class Input:
     shape: Tuple[int, ...]
-    dtype: Type
+    dtype: Any
```
Comment (jaheba) [inline src/gluonts/model/inputs.py:25]: What is the issue here?

File: src/gluonts/evaluation/metrics.py [hunk 3]
```diff
@@ -87,7 +87,7 @@ def coverage(target: np.ndarray, forecast: np.ndarray) -> float:
 
         coverage = mean(Y < \hat{Y})
     """
-    return np.mean(target < forecast)
+    return np.mean(target < forecast, dtype=float)
```
Comment (jaheba) [inline src/gluonts/evaluation/metrics.py]: Is this change related?

File: src/gluonts/model/inputs.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/model/inputs.py:25]: `torch.dtype` not recognized as a `Type` apparently

File: src/gluonts/evaluation/metrics.py (hunk 3 above)
Comment (lostella) [reply src/gluonts/evaluation/metrics.py]: Yes, let me list the issues in the OP

Comment (lostella) [reply src/gluonts/evaluation/metrics.py]:
```
src/gluonts/evaluation/metrics.py:90: error: Incompatible return value type (got "floating[Any]", expected "float")  [return-value]
```

File: src/gluonts/model/inputs.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/model/inputs.py:25]:
```
src/gluonts/torch/model/deepar/module.py:161: error: Argument "dtype" to "Input" has incompatible type "dtype"; expected "Type[Any]"  [arg-type]
```

File: src/gluonts/zebras/_period.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/zebras/_period.py]:
```
src/gluonts/zebras/_period.py:52: error: Library stubs not installed for "dateutil.parser"  [import]
src/gluonts/zebras/_period.py:52: note: Hint: "python3 -m pip install types-python-dateutil"
src/gluonts/zebras/_period.py:52: note: (or run "mypy --install-types" to install all missing stub packages)
src/gluonts/zebras/_period.py:52: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

File: src/gluonts/evaluation/metrics.py (hunk 3 above)
Comment (jaheba) [reply src/gluonts/evaluation/metrics.py]: I don't see this in the failing mypy job, e.g here: https://github.com/awslabs/gluonts/actions/runs/4355090680/jobs/7611245842

Where do you get this?

File: src/gluonts/zebras/_period.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/zebras/_period.py]: I see, but then let's install the stubs instead of doing this `type: ignore`.

Also, it doesn't currently trigger because of `--ignore-missing-imports` I think.

**Review by jaheba — APPROVED** [review]
